content/articles/how-to-add-support-for-spatial-data-in-your-database-using-geoalchemy-and-spatialite/index.md

```diff
@@ -0,0 +1,397 @@
+
+```python
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+from config import config_
+
+# initialize extensions
+db = SQLAlchemy()
+migrate = Migrate()
+
+# app factory
+def create_app(config_name):
+    # create app instance
+    app = Flask(__name__)
+
+    # configure app
+    app.config.from_object(config_[config_name])
+
+    # initialize extensions
+    db.init_app(app)
+    migrate.init_app(app, db)
+
+    return app
+
+```
+Let's add some configuration options that our flask app can use, in the `config.py` file we created earlier. So navigate to that file and add the following lines of code to add the configuration options.
+
+```python
+import os
+
+base_dir = os.path.abspath(os.getcwd())
+
+class Config:
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+    SECRET_KEY = os.getenv('app-secret')
+    # other generic configuration
+    # ..options
+
+class DevConfig(Config):
+    SQLALCHEMY_DATABASE_URI = f"sqlite:///{base_dir}/devdb.sqlite"
+    DEBUG = True
+    FLASK_ENV = 'development'
+
+class ProdConfig(Config):
+    SQLALCHEMY_DATABASE_URI = os.getenv('DATABASE_URI')
+    FLASK_ENV = 'production'
+
+config_ = {
+    'default': DevConfig,
+    'development': DevConfig,
+    'production': ProdConfig
+}
+```
+The `_config` variable in the last line holds a  dictionary that maps each config option to their names, so we can use the `config's name` to reference them when we need to add the config to our flask application. This is what we did in the `__init__.py` file, in the app factory, as shown below:
+
+```python
+# .. code ommited for brevity
+from config import config_
+# .. 
+# ...
+app.config.from_object(config_[config_name])
+```
+
+#### Running Our Flask Server
+
+To run our flask app we need to add some final instructions in the `wsgi.py` which is what will help us run the flask app. So go to that file and add the following lines of code:
+
+```python
+from core import create_app
+import os
+
+# call create_app() to create and configure flask app
+app = create_app('default' or os.getenv('FLASK_CONFIG'))
+```
+Let's also set the `FLASK_ENV` environment variable to `development` from the terminal so we don't have to restart our server each time we make a change.
+
+```bash
+FLASK_ENV=development
+```
+
+Now let's run our flask app from the command-prompt/terminal using the following command :
+
+```bash
+flask run
+```
+
+_**Note**: This command works because flask looks for any file named `wsgi.py` or `app.py` by default. If you changed your naming convention for this file, you'll have to set the FLASK_APP environment variable to the name of the file you're using._ 
```

Comment:
Instead of using `_` at the start and end. Just use `>` at the start for quoting